From e673336679ddc26881b01aaef160db1d2188c7c8 Mon Sep 17 00:00:00 2001
Message-Id: <e673336679ddc26881b01aaef160db1d2188c7c8.1431284209.git.chang-joon.lee@intel.com>
In-Reply-To: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
References: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
From: Ramalingam C <ramalingam.c@intel.com>
Date: Wed, 29 Apr 2015 13:58:09 +0530
Subject: [PATCH 56/68] Revert "REVERTME [VPG]: drm/i915: configure dual link
 pclk only for >= CHT B0"

This reverts commit 13e87787c5fbab68c145c50d34ad55c776d499ab.

As pixel overlapping feature is applicable from stepping B0 onwards,
if panel supports the feature, we can invoke the feature irrespective of
the stepping. so reverting the previous REVERTME patch.

Issue: GMINL-8115
Change-Id: I0b06dbf087f9e6485fb81450e31c28a1bf62f8e5
Signed-off-by: Ramalingam C <ramalingam.c@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi_panel_vbt.c |    8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
index 755ad96..4ecc9e2 100644
--- a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
+++ b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
@@ -742,13 +742,7 @@ static bool generic_init(struct intel_dsi_device *dsi)
 	if (intel_dsi->dual_link) {
 		pclk = pclk / 2;
 
-		/*
-		 * in case of CHT B0 and above stepping we can enable
-		 * pixel_overlap if needed by panel. In this case
-		 * we need to increase the pixelclock for extra pixels
-		 */
-		if ((IS_CHERRYVIEW(dev_priv->dev) && STEP_FROM(STEP_B0)) &&
-			(intel_dsi->dual_link & MIPI_DUAL_LINK_FRONT_BACK)) {
+		if (intel_dsi->dual_link & MIPI_DUAL_LINK_FRONT_BACK) {
 			pclk += DIV_ROUND_UP(mode->vtotal *
 					intel_dsi->pixel_overlap * 60, 1000);
 		}
-- 
1.7.9.5

